core/src/main/java/org/elasticsearch/search/highlight/HighlightBuilder.java

```diff
@@ -164,24 +169,138 @@ public Boolean useExplicitFieldOrder() {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject("highlight");
+        builder.startObject(HIGHLIGHT_ELEMENT_NAME);
         innerXContent(builder);
         builder.endObject();
         return builder;
     }
 
+    /**
+     * Creates a new {@link HighlightBuilder} from the highlighter held by the {@link QueryParseContext}
+     * in {@link org.elasticsearch.common.xcontent.XContent} format
+     *
+     * @param parseContext
+     *            the input parse context. The state on the parser contained in
+     *            this context will be changed as a side effect of this method
+     *            call
+     * @return the new {@link HighlightBuilder}
+     */
+    public static HighlightBuilder fromXContent(QueryParseContext parseContext) throws IOException {
```

Comment:
Can we not make the AbstractHighlighterBuilder define a fromXContent() method so it has the code for parsing the common attributes and then defer to a doFromXContent() to read the implementation specific attributes in HighlighterBuilder and Field? I am worried that if we duplicate this code in each implementation we will end up with parsing inconsistencies over time